guides/plugins/plugins/framework/data-handling/entities-via-attributes.md

```diff
@@ -0,0 +1,411 @@
+```
+
+## Json fields 
+
+If you want to store json data in a field, with an own validation and serialize logic, you can use the `Serialized` attribute, and define an own serializer class:
+
+```php
+#[Entity('attribute_entity')]
+class AttributeEntity extends EntityStruct
+{
+    #[PrimaryKey]
+    #[Field(type: FieldType::UUID)]
+    public string $id;
+    
+    #[Serialized(serializer: PriceFieldSerializer::class)]
+    public ?PriceCollection $serialized = null;
+}
+```
+
+## Custom Fields
+
+If you want to allow custom fields, you can use the `CustomField` attribute. 
+
+```php
+<?php
+
+#[Entity('attribute_entity')]
+class AttributeEntity extends EntityStruct
+{
+    #[PrimaryKey]
+    #[Field(type: FieldType::UUID)]
+    public string $id;
+    
+    /**
+     * @var array<string, mixed>
+     */
+    #[CustomFields]
+    public array $customFields;
+}
+
+```
+
+## Api encoding
+
+Each field of an entity is by default not exposed in the API. To expose a field in the API, you have to set the `api` parameter of the `Field` attribute to `true`, or to one of the scopes you want to allow. 
+
+```php
+<?php
+
+use Shopware\Core\Framework\Api\Context\AdminApiSource;
+use Shopware\Core\Framework\Api\Context\SalesChannelApiSource;
+
+#[Entity('attribute_entity')]
+class AttributeEntity extends EntityStruct
+{
+    #[PrimaryKey]
+    #[Field(type: FieldType::UUID, api: true)]
+    public string $id;
+
+    #[Field(type: FieldType::STRING)]
+    public string $notExposed;
+
+    #[Field(type: FieldType::STRING, api: true)]
+    public string $everywhere;
+
+    #[Field(type: FieldType::STRING, api: [AdminApiSource::class])]
+    public string $adminOnly;
+
+    #[Field(type: FieldType::STRING, api: [SalesChannelApiSource::class])]
+    public string $storeOnly;
+```
+
+## Translated fields
+
+To also support shopware translations for your entity, you can set the `translated` property of the `Field` attribute to `true`. This will automatically create a `TranslatedField` for the field and
+registers a `EntityTranslationDefinition` for you.
+
+You can also define a `Translations` attribute on a property to allow loading all translations of the entity.
+
+Notice: Properties with the `translated` flag must be nullable. 
+
+```php
+#[Entity('attribute_entity')]
+class AttributeEntity extends EntityStruct
+{
+    #[PrimaryKey]
+    #[Field(type: FieldType::UUID)]
+    public string $id;
+
+    #[Field(type: FieldType::STRING, translated: true)]
+    public ?string $string = null;
+
+    /**
+     * @var array<string, ArrayEntity>
+     */
+    #[Translations]
+    public array $translations;
+}
+```
+
+## Required fields
+
+Each field which is not type hinted as nullable is required by default. But you can also mark a field as required by adding the `Required` attribute to the field. This will automatically add a validation rule to the field.
+This is necessary for fields which are marked as `translated`, because the translated fields need to be nullable.
```

Comment:
```suggestion
By default, any field that is not type-hinted as nullable is required. However, you can explicitly mark a field as required by adding the `Required` attribute to it. This will automatically add a validation rule to the field.
This is necessary for fields marked as `translated`, as translated fields must be nullable.
```